packages/@winglang/sdk/src/target-sim/bucket.inflight.ts

```diff
@@ -27,19 +33,128 @@ import { Datetime, Json, LogLevel, TraceType } from "../std";
 
 export const METADATA_FILENAME = "metadata.json";
 
+const LOCALHOST_ADDRESS = "127.0.0.1";
+
+const STATE_FILENAME = "state.json";
+
+/**
+ * Contents of the state file for this resource.
+ */
+interface StateFileContents {
+  /**
+   * The last port used by the API server on a previous simulator run.
+   */
+  readonly lastPort?: number;
+}
+
 export class Bucket implements IBucketClient, ISimulatorResourceInstance {
   private _fileDir!: string;
   private _context: ISimulatorContext | undefined;
   private readonly initialObjects: Record<string, string>;
   private readonly _public: boolean;
   private readonly topicHandlers: Partial<Record<BucketEventType, string>>;
   private _metadata: Map<string, ObjectMetadata>;
+  private readonly app: express.Application;
+  private server: Server | undefined;
+  private url: string | undefined;
+  private port: number | undefined;
 
   public constructor(props: BucketSchema) {
     this.initialObjects = props.initialObjects ?? {};
     this._public = props.public ?? false;
     this.topicHandlers = props.topics;
     this._metadata = new Map();
+
+    this.app = express();
+    // Enable cors for all requests.
+    this.app.use((req, res, next) => {
+      const corsHeaders: CorsHeaders = {
+        defaultResponse: {
+          "Access-Control-Allow-Origin": "*",
+          "Access-Control-Allow-Methods": "GET, PUT",
+          "Access-Control-Allow-Headers": "*",
+        },
+        optionsResponse: {
+          "Access-Control-Allow-Origin": "*",
+          "Access-Control-Allow-Methods": "GET, PUT",
+          "Access-Control-Allow-Headers": "*",
+          "Access-Control-Max-Age": "86400",
+        },
+      };
+      const method =
+        req.method && req.method.toUpperCase && req.method.toUpperCase();
+
+      if (method === "OPTIONS") {
+        for (const [key, value] of Object.entries(
+          corsHeaders.optionsResponse
+        )) {
+          res.setHeader(key, value);
+        }
+        res.status(204).send();
+      } else {
+        for (const [key, value] of Object.entries(
+          corsHeaders.defaultResponse
+        )) {
+          res.setHeader(key, value);
+        }
+        next();
+      }
+    });
+    this.app.put("*", (req, res) => {
+      const action = req.query.action;
```

Comment:
Perhaps break up some of the constructor logic into helper functions like `this.addSignedUrlRoutes()`?